### PR TITLE
Feature/issues/30

### PR DESCRIPTION
**Pull request by seanl-adg**
https://github.com/AdguardTeam/ExtendedCss/issues/30
(Commits names are misspelled)

### CODE REVIEW

File: lib/extended-css-selector.js [hunk 1]
```diff
@@ -141,7 +142,7 @@ var ExtendedSelector = (function () { // jshint ignore:line
                 // If we meet complex token, all subsequent tokens are considered complex
                 // All previously found simple tokens are also considered complex
                 if (simpleTokens.length > 0) {
-                    complexTokens = complexTokens.concat(simpleTokens);
+                    Array.prototype.push.apply(complexTokens, simpleTokens);
```
Comment (atropnikov) [inline lib/extended-css-selector.js]: Why did you replace `concat`?

Comment (atropnikov) [inline lib/extended-css-selector.js]: Ah, allocation issues. Ok.

File: lib/sizzle.patched.js [hunk 2]
```diff
@@ -1572,6 +1575,8 @@ Expr = Sizzle.selectors = {
 	}
 };
 
+Expr.pseudos["has-text"] = Expr.pseudos["contains"];
```
Comment (ameshkov) [inline lib/sizzle.patched.js]: Why can't we do it on our side? Patching sizzle to add another pseudo-class seems redundant.

File: lib/extended-css-selector.js [hunk 3]
```diff
@@ -96,18 +96,17 @@ var ExtendedSelector = (function () { // jshint ignore:line
      * Joins tokens values
      */
     var joinTokens = function(selector, relationToken, tokens) {
```
Comment (ameshkov) [inline lib/extended-css-selector.js]: You might want to point out in the comment that you use string concatenation due to browsers optimizations

**Pull request merged**